include/cpp-sort/adapters/schwartz_adapter.h

```diff
@@ -44,11 +42,36 @@
 
 namespace cppsort
 {
-    ////////////////////////////////////////////////////////////
-    // Adapter
-
     namespace detail
     {
+        ////////////////////////////////////////////////////////////
+        // Dedicated comparison function
+
+        template<typename Compare>
+        class schwartz_compare
+        {
+            private:
+
+                Compare compare;
+
+            public:
+
+                explicit schwartz_compare(Compare compare):
+                    compare(compare)
+                {}
+
+                template<typename T, typename U>
+                auto operator()(T&& lhs, U&& rhs)
+                    noexcept(noexcept(compare(std::forward<T>(lhs).data, std::forward<U>(rhs).data)))
+                    -> decltype(compare(std::forward<T>(lhs).data, std::forward<U>(rhs).data))
```

Comment:
I'm sure the world would be a better place if `noexcept(auto)` was a thing and if `decltype(auto)` was actually SFINAE-friendly .___.